manage_breast_screening/assets/js/ajax-submit.js

```diff
@@ -0,0 +1,73 @@
+const TIMEOUT = 2000
+
+/*
+Enhance an HTML form to intercept submit events and instead fetch the form action URL.
+If successful, the response will be injected into the element with ID matching the data-target attribute.
+If unsuccessful, or the fetch times out, force a page refresh.
+*/
+class AjaxSubmit {
+  constructor($root) {
+    this.$root = $root
+
+    this.$form = $root.querySelector('form')
+    this.$showOnSubmit = $root.querySelectorAll('[data-show-on-submit]')
```

Comment:
does the way I've configured this with data attributes make sense? And is there any further error handling I've missed here?